src/Chapter4.hs

```diff
@@ -377,6 +384,9 @@ right? This means that functions can not only be passed as arguments
 to data types but can also be stored inside other containers. Even
 'Maybe' (e.g. "Maybe (Int -> Int)")!
 
+one question::
+A function inside a context it is possible due to pure ?
```

Comment:
Not only due to `pure`. Additionally, can put a function inside a context by partially applying some function to a value inside the context. It's possible because functions are first-class values in Haskell, and are not different from any other values.

So,for example, if you have a value `Just 4 :: Maybe Int` and a function `add :: Int -> Int -> Int`, you can do

```haskell
fmap add (Just 4)
```

to get a value of type `Maybe (Int -> Int)`. The function `add` will be partially applied to the value `4`